bibliopixel/control/api.py

```diff
@@ -0,0 +1,56 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import urllib
+import getpass, platform, sys, threading
+from .. util import log
+from . control import ExtractedControl
+
+class Api(ExtractedControl):
+    EXTRACTOR = {
+        'keys_by_type': {
+            'direct_command': ['action'],
+            'value_command': ['action', 'value']
+        }
+    }
+
+    def make_value_commands(self, value_command_list):
+        return [{'action': com, 'value': val, 'type': 'value_command'} for (com, val) in value_command_list]
```

Comment:
Quibble: You don't need the parents around `(com, val)`.  In one-liners, I often use single letter variables:

    return {'typename': 'value_command', 'action': a, 'value': v for a, v in value_command_list}

Reply:
More important, this shouldn't be a method!  Note that you don't use `self` there.  Pull it up and make it a file-level function.